opentelemetry-sdk/src/opentelemetry/sdk/_metrics/instrument.py

```diff
@@ -19,142 +19,142 @@
 # all instances. Implementations of these classes must not make any change to
 # this default dictionary in __init__.
 
+from opentelemetry._metrics.instrument import Counter as APICounter
+from opentelemetry._metrics.instrument import Histogram as APIHistogram
 from opentelemetry._metrics.instrument import (
-    Counter,
-    Histogram,
-    ObservableCounter,
-    ObservableGauge,
-    ObservableUpDownCounter,
-    UpDownCounter,
+    ObservableCounter as APIObservableCounter,
 )
-from opentelemetry.sdk._metrics.aggregation import (
-    ExplicitBucketHistogramAggregation,
-    LastValueAggregation,
-    SumAggregation,
+from opentelemetry._metrics.instrument import (
+    ObservableGauge as APIObservableGauge,
+)
+from opentelemetry._metrics.instrument import (
+    ObservableUpDownCounter as APIObservableUpDownCounter,
 )
+from opentelemetry._metrics.instrument import TCallback
+from opentelemetry._metrics.instrument import UpDownCounter as APIUpDownCounter
+from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
 
 
 class _Instrument:
     def __init__(
         self,
-        name,
-        unit="",
-        description="",
-        aggregation=None,
-        aggregation_config={},
+        instrumentation_info: InstrumentationInfo,
+        name: str,
+        *args,
+        unit: str = "",
+        description: str = "",
     ):
-        self._attributes_aggregations = {}
-        self._aggregation = aggregation
-        self._aggregation_config = aggregation_config
-        aggregation(self, **aggregation_config)
+        self._instrumentation_info = instrumentation_info
+        super().__init__(name, *args, unit=unit, description=description)
```

Comment:
What is this `super()` call for?

Reply:
Oh cool, I didn't know that interaction.

Reply:
non blocking comment

I want to suggest a more explicit way to handle all of this that doesn't couple the "sibling" parent classes. Rather than require `super().__init(...)` in Parent0 to call Parent1's constructor, you can call the individual parent class method explicitly like this

```python
class Parent0:
    def __init__(self, a, b):
        print("Parent0 init", a, b)


class Parent1:
    def __init__(self, x):
        print("Parent1 init", x)


class Child(Parent0, Parent1):
    def __init__(self, a, b, x):
        print("Child init")
        Parent0.__init__(self, a, b)
        Parent1.__init__(self, x)
```

Pros:
- Parent1 and Parent0 aren't coupled and can be used independently.
- The order of multiple parent classes in `class Child(Parent0, Parent1)` doesn't matter this way
- Constructors don't need `*args/**kwargs` to pass unknown constructor parameters to sibling classes
- Explicit. You can easily tell the order and which parameters go to which parent classes

Cons:
- Doesn't mix well with `super()` if it is used in the given method anywhere in the class hierarchy. Should probably stick to one or the other style for a given method in the hierarchy.